apollo-federation/cli/src/main.rs

```diff
@@ -295,6 +320,64 @@ fn cmd_extract(file_path: &Path, dest: Option<&PathBuf>) -> Result<(), Federatio
     Ok(())
 }
 
+fn cmd_expand(
+    file_path: &Path,
+    dest: Option<&PathBuf>,
+    filter_prefix: Option<&str>,
+) -> Result<(), FederationError> {
+    let original_supergraph = load_supergraph_file(file_path)?;
+    let ExpansionResult::Expanded { raw_sdl, .. } =
+        expand_connectors(&original_supergraph.schema.schema().serialize().to_string())?
+    else {
+        return Err(FederationError::internal(
+            "supplied supergraph has no connectors to expand",
+        ));
+    };
+
+    // Validate the schema
+    // TODO: If expansion errors here due to bugs, it can be very hard to trace
+    // what specific portion of the expansion process failed. Work will need to be
+    // done to expansion to allow for returning an error type that carries the error
+    // and the expanded subgraph as seen until the error.
+    let expanded = Supergraph::new(&raw_sdl)?;
+
+    let should_skip =
+        |name: &str| !matches!(filter_prefix, Some(prefix) if name.starts_with(prefix));
+
+    let subgraphs = expanded.extract_subgraphs()?;
+    if let Some(dest) = dest {
+        fs::create_dir_all(dest).map_err(|_| SingleFederationError::Internal {
+            message: "Error: directory creation failed".into(),
+        })?;
+        for (name, subgraph) in subgraphs {
+            // Skip any files not matching the prefix, if specified
+            if should_skip(&name) {
+                continue;
+            }
+
+            let subgraph_path = dest.join(format!("{}.graphql", name));
+            fs::write(subgraph_path, subgraph.schema.schema().to_string()).map_err(|_| {
+                SingleFederationError::Internal {
+                    message: "Error: file output failed".into(),
+                }
+            })?;
+        }
+    } else {
+        for (name, subgraph) in subgraphs {
+            // Skip any files not matching the prefix, if specified
+            if should_skip(&name) {
+                continue;
+            }
+
+            println!("[Subgraph `{}`]", name);
```

Comment:
could you print this in YAML format so we can immediately pipe it into `rover supergraph compose`?